### PR TITLE
Lesson#16 refactoring spec add shared spec

### PR DESCRIPTION
**Pull request by ValAntG**

### CODE REVIEW

File: spec/api/v1/answers_spec.rb [hunk 1]
```diff
@@ -7,70 +7,44 @@
...
 
       before { get "/api/v1/answers/#{answer.id}", params: { format: :json, access_token: access_token.token } }
 
-      it { expect(response).to be_success }
-
-      %w[id body created_at updated_at].each do |attr|
-        it { expect(response.body).to be_json_eql(answer.send(attr.to_sym).to_json).at_path(attr.to_s) }
-      end
-
       context 'with comment' do
```
Comment (svidlonix) [inline spec/api/v1/answers_spec.rb]: Is it possible https://github.com/ValAntG/Hard_course/blob/73e68266b476f87bf075a898e2a9b5167e9a6a3f/spec/api/v1/answers_spec.rb#L48-L53
take out here
https://github.com/ValAntG/Hard_course/pull/16/files#diff-72b981c01cb01ac2f90ce7664f34fbd0
?

File: spec/api/v1/questions_spec.rb [hunk 2]
```diff
@@ -5,61 +5,43 @@
...
+    it_behaves_like 'API Authenticable'
 
-      before do
-        create :question
-        get '/api/v1/questions', params: { format: :json, access_token: access_token.token }
-      end
+    it_behaves_like 'API index when authorized', %w[id title body created_at updated_at] do
+      let(:object) { question }
+      let(:size) { 2 }
+      let(:path) { '0/' }
```
Comment (svidlonix) [inline spec/api/v1/questions_spec.rb]: Thome thing is really strange.

File: spec/models/question_spec.rb [hunk 3]
```diff
@@ -8,4 +8,26 @@
...
+
+    let(:user) { create(:user) }
+
+    it 'should calculate reputation after creating' do
+      expect(Reputation).to receive(:calculate).with(question)
+      question.save!
+    end
+
+    it 'should not calculate reputation after updating' do
+      question.save!
```
Comment (svidlonix) [inline spec/models/question_spec.rb]: Why do you need it?

File: spec/support/shared/api_index_authorized.rb [hunk 4]
```diff
@@ -0,0 +1,10 @@
+shared_examples_for 'API index when authorized' do |attributes|
+  before { index_authorized }
+
+  it { expect(response).to be_successful }
+  it { expect(response.body).to have_json_size(size) }
+
+  attributes.each do |attr|
+    it { expect(response.body).to be_json_eql(object.send(attr.to_sym).to_json).at_path(path + attr.to_s) }
+  end
+end
```
Comment (svidlonix) [inline spec/support/shared/api_index_authorized.rb]: Please add the last empty line. Why rubocop does not detect it?

File: spec/api/v1/answers_spec.rb [hunk 5]
```diff
@@ -7,70 +7,44 @@
...
-        get "/api/v1/questions/#{question.id}/answers", params: { format: :json, access_token: access_token.token }
-      end
-
-      it { expect(response).to be_success }
-      it { expect(response.body).to have_json_size(2) }
-
-      %w[id body created_at updated_at].each do |attr|
-        it { expect(response.body).to be_json_eql(answer.send(attr.to_sym).to_json).at_path("0/#{attr}") }
-      end
+    def do_request(options = {})
```
[the remark this answers is not in the document]
Comment (ValAntG) [reply spec/api/v1/answers_spec.rb]: на вход принимается options.

**Pull request merged**